### PR TITLE
chore: Refactor code quality issues

### PR DESCRIPTION
**Pull request by akshgpt7**
## Description

Hey, I'm a Developer Outreach at [DeepSource](https://deepsource.io/) and ran DeepSource analysis on my fork of the repo. It found some interesting [code quality and performance improvements](https://deepsource.io/gh/akshgpt7/kombu/) to consider.

This PR fixes a few of the issues detected for you to assess if it is right for you.
Happy to provide the tweaks separately otherwise :)

## Summary of changes

- Simplify `if` statement.
- Use `sys.exit()` instead of bare `exit()`.
- Remove duplicate imports.
- Use literals to create data structures to avoid function calls.
- Add `.deepsource.toml` file for continuous analysis on bug risks/performance/code-quality issues on new changes.

### CODE REVIEW

**Comment by auvipy** [comment]
unit tests are failing

**Comment by akshgpt7** [comment]
@auvipy I've fixed the errors. Thanks!

**Review by auvipy — APPROVED** [review]

**Comment by auvipy** [comment]
@thedrow I think we can add this, what do you think?

File: .deepsource.toml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    "extra/**",
+    "examples/**"
+]
+
+[[analyzers]]
+name = "python"
+enabled = true
+
+  [analyzers.meta]
+  runtime_version = "3.x.x"
```
Comment (matusvalo) [inline .deepsource.toml:16]: I think this file should go in separate PR. There should be decision that we want to use deepsource and it should be configured to trigger it by PR.

File: kombu/asynchronous/http/curl.py [hunk 2]
```diff
@@ -266,7 +266,8 @@ def ioctl(cmd):
             else:
                 setopt(_pycurl.INFILESIZE, len(body))
         elif request.method == 'GET':
-            assert not request.body
+            if request.body:
+                raise AssertionError
```
Comment (matusvalo) [inline kombu/asynchronous/http/curl.py:270]: Why is raising `AssertionError` better than using `assert` keyword?

**Review by matusvalo — CHANGES_REQUESTED** [review]

File: kombu/asynchronous/http/curl.py (hunk 2 above)
Comment (akshgpt7) [reply kombu/asynchronous/http/curl.py:270]: Python has an option to compile the optimized bytecode and create the respective .pyo files by using the options `-O` and `-OO`.

During this, these basic optimizations are done:
- All the assert statements are removed
- All docstrings are removed (when `-OO` is selected)
- Value of the debug built-in variable is set to False

Now, if someone never intends to run the optimized bytecode, they can ignore this issue. But, it is still recommended not to have it in non-test files.
A better way for internal self-checks is to explicitly check and raise respective Error using an if statement.

File: .deepsource.toml (hunk 1 above)
Comment (akshgpt7) [reply .deepsource.toml:16]: No worries, I'll remove the file.
Do you want me to open an issue for discussion regarding activating DeepSource?

Comment (thedrow) [reply .deepsource.toml:16]: @akshgpt7 Please do.
I'm using DeepCode in another project.
Are they a direct competitor of yours?

File: kombu/asynchronous/http/curl.py (hunk 2 above)
Comment (thedrow) [reply kombu/asynchronous/http/curl.py:270]: I strongly disagree.

If we do decide to make this part of the official contract, we need to raise a proper exception (in this case, a ValueError with an appropriate message).
If we need this during development only, the assert statement should stay and be optimized out of the bytecode in production.

**Comment by akshgpt7** [comment]
@matusvalo any updates on this?

**Comment by thedrow** [comment]
@akshgpt7 Please address the code review comments.

File: .deepsource.toml (hunk 1 above)
Comment (akshgpt7) [reply .deepsource.toml]: @thedrow There are various added benefits that DeepSource offers over other code analysis tools like DeepCode and Codacy:
- Automatic code formatting to comply with the code style (through Transformers).
- Option to Autofix detected issues through one click.
- DeepSource Python analyzer detects 520 issues as of now (new support being continuously added).
- Less than 5% false positive reports in DeepSource.
- Option to report false positives directly to the team.

These are of course some of the benefits, just for you to assess if the tool is right for the project :)
If you have any questions, feel free to reach out or have a look at the [docs](https://deepsource.io/docs).

Comment (akshgpt7) [reply .deepsource.toml]: Also, I've removed the config file, and opened an issue #1329.

File: kombu/asynchronous/http/curl.py (hunk 2 above)
Comment (akshgpt7) [reply kombu/asynchronous/http/curl.py:270]: Do you want me to revert this change then?

Comment (thedrow) [reply kombu/asynchronous/http/curl.py:270]: Either that or transform them into proper exceptions.

**Review by thedrow — APPROVED** [review]
